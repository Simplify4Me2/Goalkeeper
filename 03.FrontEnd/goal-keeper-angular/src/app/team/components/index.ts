export * from './team-info/team-info.component';
export * from './team-players/team-players.component';
export * from './team-goals-stats/team-goals-stats.component';
export * from './team-cards-stats/team-cards-stats.component';
export * from './team-charts/team-charts.component';
export * from './player-card/player-card.component';