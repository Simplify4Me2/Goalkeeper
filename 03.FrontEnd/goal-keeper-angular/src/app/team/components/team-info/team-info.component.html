<div *ngIf="team" class="team" >
    <h1 class="team-name" >{{ team.name }}</h1>
    <div class="team-background" >
        <img class="logo" src='{{ "assets/images/logos/" + team.name + ".svg" }}' />
        <div class="right-content" >
            <div></div>
            <div class="team-form" >
                <span class="form-span">Form guide:</span>
                <span class="form-streak win">W</span>
                <span class="form-streak draw">D</span>
                <span class="form-streak win">W</span>
                <span class="form-streak loss">L</span>
            </div>
            <span class="stadium-name">{{ mockStadiumName }}</span>
        </div>
    </div>
    <!-- <img class="stadium" src="https://www.rsca.be/sites/rsca/files/2019-10/Lotto%20Park_0.jpg" /> -->
    <!-- <div class="team-name">
    </div> -->
    <!-- <img src="https://resources.ohleuven.com/oud-heverlee-leuven/photo/2018/08/29/516f4860-40c9-4fb6-b4ca-de06803daf3e/775209246-014.jpg" />
    <img src="https://www.rsca.be/sites/rsca/files/2019-10/Lotto%20Park_0.jpg" /> -->
</div>
