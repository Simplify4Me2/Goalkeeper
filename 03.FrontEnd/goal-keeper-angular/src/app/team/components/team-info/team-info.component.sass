// @import 'styles'
// @import '../../../../sass/_variables'
// @use 'styles'
// @use '../../../../styles'
// @use '~/s'
// @use '_variables'
// @use 'src/app/sass/_variables'
@use '../../../../sass/_variables' as var

$antwerp-rgba: rgba(223, 23, 43, 0.75)
$antwerp-stadium: '/assets/images/stadiums/antwerp.jpg'
$anderlecht-stadium: 'https://www.rsca.be/sites/rsca/files/2019-10/Lotto%20Park_0.jpg'
    
// .stadium
//     display: block
//     width: 40%
//     margin-left: auto
//     margin-right: auto
//     padding: 1rem
//     border-width: 5px
//     border-style: solid
//     border-image: linear-gradient(to bottom, red, rgba(0,0,0,0)) 1 100%

.team
    // font-family: 'Permanent Marker', Charcoal, sans-serif
    // font-family: $primary-font-family
    font-family: var.$primary-font-family
    // font-family: _variables.$primary-font-family

.team-name
    text-align: center

.logo
    height: 70%
    padding-left: 2rem

    // https://css-tricks.com/tinted-images-multiple-backgrounds/
.team-background
    height: 260px
    width: 100%
    margin-bottom: 40px
    background: linear-gradient($antwerp-rgba, $antwerp-rgba), url($antwerp-stadium) no-repeat center center
    background-size: 100% 200%

    display: flex
    justify-content: space-between
    align-items: center

.right-content
    display: flex
    flex-direction: column
    padding-right: 1rem
    height: 100%
    justify-content: space-between

.team-form
    font-family: var.$secondary-font-family
    height: 88px
    width: 352px
    background: white
    border: 2px solid #FF0000
    display: flex
    justify-content: space-between
    align-items: center
    padding: 0rem 1rem

.stadium-name
    color: white
    align-self: flex-end

.form-span
    font-weight: bold
    font-size: 16px

.form-streak
    font-weight: bold
    font-size: 36px

.win
    color: green

.loss
    color: red
