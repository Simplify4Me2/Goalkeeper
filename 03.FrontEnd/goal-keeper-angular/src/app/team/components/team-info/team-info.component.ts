import {
  ChangeDetectionStrategy,
  Component,
  Input,
  OnInit,
} from '@angular/core';
import { Team } from '../../models';

@Component({
  selector: 'app-team-info',
  templateUrl: './team-info.component.html',
  styleUrls: ['./team-info.component.sass'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class TeamInfoComponent implements OnInit {
  @Input() team: Team;

  constructor() {}

  ngOnInit(): void {
  }

  getTeamBackground(teamName: string) {
    // return {
    //   btn: true,
    //   'btn-primary': true,
    //   'btn-extra-class': this.stateFlag,
    // };
    return {
      'team-background': true,
      'antwerp': teamName === 'Royal Antwerp FC',
      'anderlecht': teamName === 'RSC Anderlecht',
      'beerschot': teamName === 'Beerschot VA',
      'cercle-brugge': teamName === 'Cercle Brugge',
      'charleroi': teamName === 'Royal Charleroi Sporting Club',
      'club-brugge': teamName === 'Club Brugge',
      'eupen': teamName === 'KAS Eupen',
      'genk': teamName === 'KRC Genk',
      'gent': teamName === 'KAA Gent',
      'kortrijk': teamName === 'KV Kortrijk',
      'mechelen': teamName === 'KV Mechelen',
      'ohl': teamName === 'Oud-Heverlee Leuven',
      'oostende': teamName === 'KV Oostende',
      'seraing': teamName === 'RFC Seraing',
      'standard': teamName === 'Standard Luik',
      'stvv': teamName === 'Sint-Truidense VV',
      'union': teamName === 'Union',
      'zulte-waregem': teamName === 'SV Zulte Waregem'
    };
  }
}
