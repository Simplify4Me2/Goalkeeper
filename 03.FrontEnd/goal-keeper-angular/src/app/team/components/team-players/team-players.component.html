<ng-container *ngIf="players">
  <div class="position-title">
    <!-- https://css-tricks.com/line-on-sides-headers/ -->
    <h3><span>Goalkeepers</span></h3>
    <section>
      <div *ngFor="let player of players | filterByPosition: 'GK'">
        <app-player-card [player]="player"></app-player-card>
      </div>
    </section>
  </div>
  <div class="position-title">
    <h3><span>Defenders</span></h3>
    <section>
      <div *ngFor="let player of players | filterByPosition: 'DEF'">
        <app-player-card [player]="player"></app-player-card>
      </div>
    </section>
  </div>
  <div class="position-title">
    <h3><span>Midfielders</span></h3>
    <section>
      <div *ngFor="let player of players | filterByPosition: 'MID'">
        <app-player-card [player]="player"></app-player-card>
      </div>
    </section>
  </div>
  <div class="position-title">
    <h3><span>Attackers</span></h3>
    <section>
      <div *ngFor="let player of players | filterByPosition: 'ATT'">
        <app-player-card [player]="player"></app-player-card>
      </div>
    </section>
  </div>
</ng-container>
