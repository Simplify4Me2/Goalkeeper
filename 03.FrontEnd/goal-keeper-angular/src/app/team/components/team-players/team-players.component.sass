@use '../../../../sass/_variables' as var

.players-table
  width: 100%
  text-align: center

table
  font-family: sans-serif
  th, td
    padding: .25em .5em
    text-align: left
    &:nth-child(2)
      text-align: right
  td
    background-color: #eee
  th
    background-color: #009
    color: #fff

  // https://codepen.io/chris22smith/pen/NWWRWxJ
// .zigzag 
//   border-collapse:separate
//   border-spacing:.25em 1em
//   tbody tr:nth-child(odd) 
//     transform:rotate(2deg)
//   thead tr,
//   tbody tr:nth-child(even) 
//     transform:rotate(-2deg)

.position-title
  font-family: var.$secondary-font-family
  font-size: 36px
  line-height: 42px
  display: flex
  justify-content: center

.position-title span:before
.position-title span:after
  background-color: red

.player-section
  display: flex
  flex-direction: column

.player-card-left
  display: flex
  flex-direction: column

.player-card-right
  display: flex
  flex-direction: column
// .position-title hr
//   // content: '              yolo  '
//   text-decoration: line-through
  // border-top: 1px solid red

// .position-title span:before,
// .position-title span:after 
//   content: ""
//   position: absolute
//   height: 5px
//   border-bottom: 1px solid red
//   border-top: 1px solid red
//   top: 0
//   width: 600px

// .position-title span:before 
//   right: 100%
//   margin-right: 15px

// .position-title span:after 
//   left: 100%
//   margin-left: 15px

