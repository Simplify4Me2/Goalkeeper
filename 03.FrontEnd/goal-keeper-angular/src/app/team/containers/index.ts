export * from './team.container';
export * from './matches/matches.container';
export * from './squad/squad.container';
export * from './stats/stats.container';