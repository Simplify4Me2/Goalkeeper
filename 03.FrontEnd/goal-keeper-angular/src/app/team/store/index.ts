export * from './team.reducer';
export * from './team.actions';
export * from './team.selectors';