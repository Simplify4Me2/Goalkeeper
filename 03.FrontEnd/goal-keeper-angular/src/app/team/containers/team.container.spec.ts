import { ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { MockStore, provideMockStore } from '@ngrx/store/testing';
import { ActivatedRoute, convertToParamMap } from '@angular/router';
import { BehaviorSubject } from 'rxjs';

import * as fromActions from '../store/team.actions';
import { TeamContainerComponent } from '.';
// import { TeamComponent } from './team.container';
import { TeamInfoComponent } from '../components/team-info/team-info.component';
import { TeamPlayersComponent } from '../components/team-players/team-players.component';
import { selectTeam } from '../store';

describe('Team Container', () => {
    let fixture: ComponentFixture<TeamContainerComponent>;
    let instance: TeamContainerComponent;
    let store: MockStore;

    beforeEach(() => {
        TestBed.configureTestingModule({
            imports: [RouterTestingModule],
            declarations: [TeamContainerComponent, TeamInfoComponent, TeamPlayersComponent],
            providers: [
                {
                    provide: ActivatedRoute,
                    useValue: {
                        snapshot: {
                            paramMap: convertToParamMap({
                                id: '112'
                            })
                        }
                    }
                },
                provideMockStore()
            ]
        });

        fixture = TestBed.createComponent(TeamContainerComponent);
        instance = fixture.componentInstance;
        store = TestBed.inject(MockStore);

        spyOn(store, 'dispatch');
    });

    it('should compile', () => {
        fixture.detectChanges();

        expect(fixture).toMatchSnapshot();
    });

    it('should dispatch a getTeam on init', () => {
        const action = fromActions.getTeam({ id: 112 });

        fixture.detectChanges();

        expect(store.dispatch).toHaveBeenCalledWith(action);
    });

    // it('should dispatch a getPlayers on init', () => {
    //     const action = getPlayers({ teamId: 112 });

    //     fixture.detectChanges();

    //     expect(store.dispatch).toHaveBeenLastCalledWith(action);
    // })
})