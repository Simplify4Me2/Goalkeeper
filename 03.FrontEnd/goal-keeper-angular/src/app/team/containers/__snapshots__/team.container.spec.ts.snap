// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team Container should compile 1`] = `
<app-team
  route={[Function Object]}
  store={[Function MockStore]}
  team={[Function Store]}
>
  <h1>
    Team container
  </h1><app-team-info>
    
  </app-team-info><app-team-players>
    <p>
      team-players works!
    </p>
  </app-team-players>
</app-team>
`;
